src/CellData/SkeletonCellFieldPair.jl

```diff
@@ -0,0 +1,98 @@
+"""
+  SkeletonCellFieldPair is a special construct for allowing uh.plus
+  and uh.minus to be two different CellFields. In particular, it is
+  useful when we need one of the CellFields to be the dualized
+  version of the other for performing ForwardDiff AD of a Skeleton
+  integration DomainContribution wrt to the degrees of freedom
+  of the CellField, plus and minus sensitivities done separately,
+  so as to restrict the interaction between the dual numbers.
+
+  It takes in two CellFields and stores plus version of CellFieldAt
+  for the first CellField and minus version of CellFieldAt for the
+  second the CellField. SkeletonCellFieldPair is associated with the
+  same triangulations as the CellFields
+"""
+struct SkeletonCellFieldPair{
+  P<:CellField, M<:CellField, T<:Triangulation} <: CellField
+
+  cf_plus::P
+  cf_minus::M
+  trian::T # Background Triangulation to which CellFields belong to
+
+  function SkeletonCellFieldPair(
+    cf_plus_plus::CellFieldAt,
+    cf_minus_minus::CellFieldAt,
+    trian::Triangulation)
```

Comment:
I would remove the `trian` argument from the inner constructor. In practice, we always extract it from either cf_plus_plus/ or cf_minus_minus.

Reply:
regarding having `trian`  in the `SkeletonCellFieldPair`, i though it can provide quick access rather than go over `getproperty` multiple times.

Reply:
Ok. We can keep it as a member variable of `SkeletonCellFieldPair`, but we do not need to have it as an argument to the constructor. This is dangerous as it opens a door to pass a `Triangulation` that has nothing to do with the one underlying the first two arguments.

Reply:
yeah true, I ll make the changes and move the outer constructor as the inner constructor

Reply:
Hi Alberto, made changes, please check..

Reply:
> Hi Alberto, made changes, please check..

see https://github.com/gridap/Gridap.jl/pull/797#discussion_r895264561